lib/salus/scanners/base.rb

```diff
@@ -124,64 +125,263 @@ def report_dependency(file, hsh = {})
+          when :file
+            create_file_option(
+              keyword: keyword,
+              value: value,
+              prefix: prefix,
+              suffix: suffix,
+              separator: separator
+            )
+          end
+        end
+      when :list
+        create_list_option(
+          keyword: keyword,
+          value: value,
+          prefix: prefix,
+          separator: separator,
+          suffix: suffix,
+          regex: regex,
+          join_by: join_by
+        )
+      when :list_file, :file_list
+        create_list_file_option(
+          keyword: keyword,
+          value: value,
+          prefix: prefix,
+          separator: separator,
+          suffix: suffix,
+          join_by: join_by
+        )
+      else
+        report_warn(
+          :scanner_misconfiguration,
+          "Could not interpolate config #{keyword_string} \
+          the type of #{type}. "
+        )
+        '' # Return an empty string and warn
+      end
+    end
+
+    def build_options(prefix:, suffix:, separator:, args:, join_by: ',')
+      default_regex = /.*/
+      args.reduce('') do |options, (keyword, type_value)|
+        keyword_string = keyword.to_s
+        option = if @config.key?(keyword_string)
+                   config_value = @config.fetch(keyword_string)
+                   case type_value
+                   when Symbol, String
+                     build_option(
+                       prefix: prefix,
+                       suffix: suffix,
+                       separator: separator,
+                       type: type_value,
+                       keyword: keyword_string,
+                       value: config_value,
+                       join_by: join_by,
+                       regex: default_regex
+                     )
+                   when Hash # If you are doing something complicated
+                     if type_value[:type].nil?
+                       warning = "Could not interpolate config \
+                       defined by since there was no type defined in the hash "
+                       report_warn(:scanner_misconfiguration, warning)
+                       '' # Return an empty string and warn
+                     else
+                       build_option(
+                         prefix: type_value[:prefix] || prefix,
+                         suffix: type_value[:suffix] || suffix,
+                         separator: type_value[:separator] || separator,
+                         keyword: type_value[:keyword] || keyword_string,
+                         value: config_value,
+                         type: type_value[:type],
+                         regex: type_value[:regex] || default_regex,
+                         join_by: type_value[:join_by] || join_by
+                       )
+                     end
+                   when Regexp # Assume it is a string type if just regex is supplied
```

Comment:
Is this true? Can I provide a regex against file matching?

*.js for example?

Reply:
Yeah, you could, but it would need to be /.*\.js/

The only difference between a string and a file type in these is whether or not the code checks if the file already exists.